models/User.js

```diff
@@ -0,0 +1,41 @@
+const Mongoose = require('mongoose');
+const bcrypt   = require('bcrypt-nodejs');
+
+const UserSchema = Mongoose.Schema({
+    username: {
+        type: String,
+        required: true,
+        index: true
+    },
+    password: {
+        type: String,
+        required: true
+    },
+    email: {
+        type: String,
+        required: true
+    },
+    address: {
+        type: String,
+        required: true
+    },
+    fullname: {
+        type: String,
+        required: true
+    },
+    dateCreated:{
+        type: Date,
+        default: Date.now()
+    }
+});
+
+UserSchema.pre('save', function(next) {
+  this.password = bcrypt.hashSync(this.password, bcrypt.genSaltSync(8), null);
```

Comment:
Nice job going the extra mile here. We will add this as a requirement for future versions of the course :D